### ISSUE TITLE
Empty matrix with MLDouble

### ISSUE THREAD

**Issue by akshaybabloo** [gradusnikov/jmatio#2]
Hi,

I have tried creating an empty matrix of 0x1 or 0x0, but I JMatIO always tends to give an output of 1x0.
when i do this
```
String data_str = "data";
double[] data_1 = new double[]{}; // also tried new double[0]
MLDouble data1 = new MLDouble(data_str, data_1, 1);
```
Is there any alternate to this?

thanks.

**Comment by kjohnston82** [comment]
I don’t think you are going to be able to do this with the currently method in place (MLDouble.java). Looking at the source code: 
------------------------------------
    /**
     * Converts double[][] to Double[]
     * 
     * @param dd
     * @return
     */
    private static Double[] double2DToDouble ( double[][] dd )
    {
        Double[] d = new Double[ dd.length*dd[0].length ];
        for ( int n = 0; n < dd[0].length; n++ )
        {
            for ( int m = 0; m < dd.length; m++ )
            {
                d[ m+n*dd.length ] = dd[m][n]; 
            }
        }
        return d;
    }
----------------------------------------

So the expectation is that dd[0].length exists in order to make the Double. So two things: (1) we might want to think about adding a throw into the code here, (2) why exactly are you interested in saving an empty matrix to matlab?


Kyle

**Comment by akshaybabloo** [comment]
I wanted to create a structure which has an empty matrix, this structure is based on a template which has something like this `dataset.data = []` this creates an empty variable to dataset structure, which is further used in the programme. and the other reason is when you use `cat` in Matlab to concatenate two `1x0`, I get an error but where as `0x1` or `0x0` doesn't.
